### PR TITLE
Modify templates to reference Mvc.NewtonsoftJson

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Templating/build.cmd [hunk 1]
```diff
@@ -0,0 +1,3 @@
+@ECHO OFF
+SET RepoRoot=%~dp0..\..
+%RepoRoot%\build.cmd -projects %~dp0\**\*.*proj %*
```
Comment (natemcmaster) [inline src/Templating/build.cmd]: These build.sh/cmd scripts won't work until #4246 is done.

File: src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/WebApi-CSharp.csproj.in [hunk 2]
```diff
@@ -15,4 +15,8 @@
     <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureADB2C.UI" Version="${MicrosoftAspNetCoreAuthenticationAzureADB2CUIPackageVersion}" Condition="'$(IndividualB2CAuth)' == 'True'" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (natemcmaster) [inline src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/WebApi-CSharp.csproj.in]: Let's condense item groups. You can merge this with the item group above and remove the Condition.

Comment (natemcmaster) [reply src/Templating/src/Microsoft.DotNet.Web.ProjectTemplates/WebApi-CSharp.csproj.in]: Looks like that's true of some of the other projects as well.

**Comment by natemcmaster** [comment]
> Running tests locally currently fails because it cannot restore the package. Do templates use packages published to MyGet?

Yes, but we're working on a new model for this repo which doesn't require building packages and restoring your own packages in order to build. #4246. Templating needs to be updated to use ProjectRefs instead, but we're not quite done with its dependencies.

**Review by rynowak — APPROVED** [review]
For the code/reference changes

**Review by pranavkm — COMMENTED** [review]
🆙 📅

**Pull request merged**